### ISSUE TITLE
[Bug]: Different LMM results on Mac vs PC (and vs older version)

### ISSUE THREAD

**Issue by iandstephen** [jasp-stats/jasp-issues#3051]
### JASP Version

0.19.1

### JASP Module

Mixed Models

### What analysis are you seeing the problem on?

Linear Mixed Model

### What OS are you seeing the problem on?

Windows 11, macOS Silicon

### Bug Description

Running a Linear Mixed Model using one DV, one continuous Fixed Factor and one Random Factor. I am getting 3 different sets of results on v17.2 compared with v19.1 on a PC compared with v19.1 on a Mac. All options, data etc are identical.

v17.2 results:
![image](https://github.com/user-attachments/assets/b6635592-d55f-4425-b4f2-bd66c7838406)
![image](https://github.com/user-attachments/assets/8f7b0914-f42c-4296-9c5c-74b856119d39)
![image](https://github.com/user-attachments/assets/d05b2c89-a495-4500-a3c9-40a1a3e66ad6)

v19.1 results on Mac:
![image](https://github.com/user-attachments/assets/aabd16f5-91f3-4d8e-9dbb-fea5001bab99)

v19.1 results on PC:
![image](https://github.com/user-attachments/assets/3e519c2e-d4a3-454d-a6c9-7a70fe441aae)
![image](https://github.com/user-attachments/assets/9952929c-dae6-4fb0-9109-179656df1e51)

Data file as .csv
[dragons.csv](https://github.com/user-attachments/files/17950086/dragons.csv)

### Expected Behaviour

Produce all the same results.

### Steps to Reproduce

1.
2.
3.
...

### Final Checklist

- [X] I have included a screenshot showcasing the issue, if possible.
- [ ] I have included a JASP file (zipped) or data file that causes the crash/bug, if applicable.
- [ ] I have accurately described the bug, and steps to reproduce it.

**Comment by tomtomme** [comment]
@iandstephen 
thx for the report. Satterthwaite is not described in the help file, but I guess it is based on some kind of bootstrap?
Setting the seed to a fixed number should give repeatable results

**Comment by FBartos** [comment]
Hi @iandstephen ,

Satterwaite is not based on a bootstrap so that should not be an issue.
It is a shame that you don't have a screenshot of the settings from 17.2 or the note from bellow the table --- the best available guess is that the random structure is not the same (as the overall F-test already differs, and one of the models clearly did not converge).

Frantisek

**Comment by iandstephen** [comment]
Hi @tomtomme I did look for bootstrap options but couldn't see any being used. These results are also repeatable within the version and OS - I was using it as a workshop exercise in a stats class and all the Macs were getting one result, all the lab PCs (which run the older version) were getting another result and all the laptop PCs (who had installed the newest version) were getting a third result.

@FBartos here is the setup and full results from the older version (v18.3 in this case but giving the same results as v17.2)
![image](https://github.com/user-attachments/assets/b1d25d5c-25de-426c-a496-15b632ec7393)
![image](https://github.com/user-attachments/assets/325fdd41-4407-4ff9-a957-d4a1ba477d77)

**Comment by FBartos** [comment]
@iandstephen , so, just to check that I understand well -- when you used the same settings in v.17 and v.18 , you verified you got the same results, right? So that should be correct.
And for the previous version you don't have the settings to verify, right?

**Comment by FBartos** [comment]
Thanks! this is very detailed and very helpful! I just tried reproducing the results and I get the same results with 19.2 as you get with pre 18.3 and 17.2 but I also get the same results as you get with 19.1 with the 19.0 version = so, I also observe the difference

**Comment by FBartos** [comment]
The most likely reason for the difference is the fact the the model is singular. This means that the optimization algorith did not converge and it can be unstable. I did try re-running the results a couple of times and they seem to be the same within a version, however, the across version stability for singular models is extremely hard to achive. There are different R-versions, different Rtools versions, different package versions etc..., small changes can easily impact the numerical stability and singular models can result in different estimates. 

In both versions (19.2 and 19.0 which differ in the full model you specified), I tried simplifying the random effects structure by a) removing correlation between the random intercepts and the random slope and b) removing the random slope al together. Both changes lead to the same output between 19.2 and 19.0.

 I'm sorry, but it seems like there is nothing else we can do about this as this behavior is most likely a result of singular model fit unstability.
